Spanner/tests/System/PgBatchTest.php

```diff
@@ -0,0 +1,157 @@
+                decade INTEGER NOT NULL
+            )',
+            self::$tableName
+        ))->pollUntilComplete();
+
+        if (self::$database->info()['databaseDialect'] == DatabaseDialect::POSTGRESQL) {
+            self::$database->updateDdlBatch([
+                sprintf(
+                    'CREATE ROLE %s',
+                    self::$dbRole
+                ),
+                sprintf(
+                    'CREATE ROLE %s',
+                    self::$restrictiveDbRole
+                ),
+                sprintf(
+                    'GRANT SELECT(id) ON TABLE %s TO %s',
+                    self::$tableName,
+                    self::$restrictiveDbRole
+                ),
+                sprintf(
+                    'GRANT SELECT ON TABLE %s TO %s',
+                    self::$tableName,
+                    self::$dbRole
+                )
+            ])->pollUntilComplete();
+        }
+
+        self::seedTable();
+    }
+
+    private static function seedTable()
+    {
+        $decades = [1950,1960,1970,1980,1990,2000];
+        for ($i = 0; $i < 250; $i++) {
+            self::$database->insert(self::$tableName, [
+                'id' => self::randId(),
+                'decade' => array_rand($decades)
+            ], [
+                'timeoutMillis' => 50000
+            ]);
+        }
+    }
+
+    public function testBatchWithDbRole()
+    {
+        // Emulator does not support FGAC
+        $this->skipEmulatorTests();
+
+        foreach ($this->dbProvider() as $dbProvided) {
+            list($dbRole, $expected) = $dbProvided;
+            $error = null;
+            $query = 'SELECT
+                    id,
+                    decade
+                FROM ' . self::$tableName . '
+                WHERE
+                    decade > $1
+                AND
+                    decade < $2';
+
+            $parameters = [
+                'p1' => 1960,
+                'p2' => 1980
+            ];
+
+            $resultSet = iterator_to_array(self::$database->execute($query, ['parameters' => $parameters]));
+
+            $batch = self::$client->batch(self::INSTANCE_NAME, self::$dbName, ['databaseRole' => $dbRole]);
+            $string = $batch->snapshot()->serialize();
+
+            $snapshot = $batch->snapshotFromString($string);
+
+            try {
+                $partitions = $snapshot->partitionQuery($query, ['parameters' => $parameters]);
+            } catch (ServiceException $e) {
+                $error = $e;
+            }
+
+            if ($expected === null) {
+                $this->assertEquals(count($resultSet), $this->executePartitions($batch, $snapshot, $partitions));
+            } else {
+                $this->assertInstanceOf(ServiceException::class, $error);
+                $this->assertEquals($error->getServiceException()->getStatus(), $expected);
+            }
+            $snapshot->close();
+        }
+    }
+
+    private function dbProvider()
```

Comment:
Same for `dbProvider`, it can be abstracted out to save from repetition.

Reply:
Done.